Recent glibc now includes strl* functions

In 032c6b8b several Linux-specific fixes were added, including implementations of `strlcpy()` and `strlcat()`, which at the time only existed in BSD-derived OSes. These are now [part of POSIX](https://www.austingroupbugs.net/view.php?id=986) and, as of version 2.38, included in glibc. This means recent Linux distros could hit the following compilation error:

```
fqd_private.h:101:22: error: static declaration of 'strlcpy' follows non-static declaration
  101 | static inline size_t strlcpy(char *dst, const char *src, size_t size)
      |                      ^~~~~~~
In file included from fq_client.c:28:
/usr/include/string.h:506:15: note: previous declaration of 'strlcpy' with type 'size_t(char * restrict,  const char * restrict,  size_t)' {aka 'long unsigned int(char * restrict,  const char * restrict,  long unsigned int)'}
  506 | extern size_t strlcpy (char *__restrict __dest,
      |               ^~~~~~~
```

We should attempt to detect the existence of these functions and only declare our own if they do not exist.

--- a/fqd_private.h
+++ b/fqd_private.h
@@ -98,6 +98,7 @@ void fq_thread_setname(const char *format, ...);
 remote_anon_client *fqd_ccs_dequeue_work(void);
 
 #if defined(linux) || defined(__linux) || defined(__linux__)
+#if ((__GLIBC__ == 2) && (__GLIBC_MINOR__ < 38))
 static inline size_t strlcpy(char *dst, const char *src, size_t size)
 {
   if(size > 0) {
@@ -121,6 +122,7 @@ static inline size_t strlcat(char *dst, const char *src, size_t size)
 
   return dl+strlen(src);
 }
-#endif
+#endif /* glibc */
+#endif /* linux */
 
 #endif

